src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpBasicFactory.php

```diff
@@ -77,7 +77,7 @@ public function addConfiguration(NodeDefinition $node)
         ;
     }
 
-    protected function createEntryPoint(ContainerBuilder $container, string $id, array $config, ?string $defaultEntryPoint)
+    public function createEntryPoint(ContainerBuilder $container, string $id, array $config, ?string $defaultEntryPoint): string
```

Comment:
This seems like a BC break for child implementations :/ as signatures become incompatible because of the different visibility level + return type. See https://3v4l.org/AmmU5

Reply:
Hmmm, good point (although I'm unsure how many people are extending these classes, but anywY they are not internal/final).

There is also the unused `$defaultEntryPoint` argument in the interface, which I hoped would fix the compatibility. But I think the only solution is coming up with a new method name in the entry point interface. `createAuthenticationEntryPoint`, `registerEntryPoint`, ...?

Reply:
These classes should really be considered final. Can we mark them as such with the `@final` annotation?

Reply:
Works for me with a CHANGELOG entry telling that all those factories have been made `final`

Reply:
Officially, the BC promise says "Classes that received the @final annotation after their first release are considered final in their next major version". Should we stick to that, or should we make them final now and consider them final now?

Reply:
I know, but this is (probably) a case where these classes *should* have been final from the get go. Let's see what others think? /cc @symfony/mergers

Reply:
As 4 core team members have voted in favor of this change, I've now marked all factories final.